DogScepterLib/Core/GMData.cs

```diff
@@ -4,109 +4,195 @@
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Release = 0;
+
+        /// <summary>
+        /// The build version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Build = 0;
+
+        /// <summary>
+        /// Indicates the bytecode format of the GameMaker data file.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.
+        /// For example, this has a value of <c>17</c> from version 2.2.1 since at least 2022.3. </remarks>
+        public byte FormatID = 0;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns chunks to 16 bytes.
+        /// </summary>
+        /// <remarks>GameMaker versions after Game Maker: Studio 2.2.2 use this. On older versions,
+        /// chunks are 4 byte aligned instead.</remarks>
+        public bool AlignChunksTo16 = true;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns strings to 4 bytes.
+        /// </summary>
+        public bool AlignStringsTo4 = true;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns backgrounds to 8 bytes.
+        /// </summary>
+        public bool AlignBackgroundsTo8 = true;
+
+        /// <summary>
+        /// TODO: Whether the GameMaker data file uses Pre-Create events for rooms and objects.
+        /// </summary>
+        public bool RoomObjectPreCreate = false;
+
+        /// <summary>
+        /// Whether the GameMaker data file contains various variable counts whose purpose is currently unknown.
+        /// </summary>
+        public bool DifferentVarCounts = false;
+
+        /// <summary>
+        /// Whether the GameMaker data file uses option flags in the OPTN chunk.
+        /// </summary>
+        public bool OptionBitflag = true;
+
+        /// <summary>
+        /// Indicates whether this GameMaker data file was run from the IDE.
+        /// </summary>
+        public bool RunFromIDE = false;
+
+        /// <summary>
+        /// Whether the audio group is builtin to the GameMaker data file.
+        /// </summary>
+        //TODO: shouldn't this be a bool, considering it can only hold values 0 and 1 and that at a low level those two are the same?
+        public int BuiltinAudioGroupID => (Major >= 2 || (Major == 1 && (Build >= 1354 || (Build >= 161 && Build < 1000)))) ? 0 : 1;
+
+        /// <summary>
+        /// Sets the recorded version number (<see cref="Major"/>, <see cref="Minor"/>, <see cref="Release"/>, <see cref="Build"/>), but
+        /// only if the parameters are higher..
+        /// </summary>
+        public void SetVersionNumber(int major = 1, int minor = 0, int release = 0, int build = 0)
```

Comment:
I originally omitted "Version" from this function name because of the verbosity in every location this is called. I'm not sure if it's better or not, but if this is kept, we also should rename IsNumberAtLeast for consistency, which would be even _more_ verbose.

Reply:
Hmm, what about changing both to ´SetVersion` and ´IsVersionAtLeast` instead then? That way it should still be clear enough that it's specifically for the version instead of some generic number.

Reply:
Yeah, that could work.